src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -1212,23 +1260,53 @@ public String getNextPage() {
             return nextPage;
         }
     }
+    
+    private class PagedAsyncListCompletionHandler<T> extends PagedAsyncCompletionHandler<List<T>> {
+        final Class<T> clazz;
+        final String name;
+        public PagedAsyncListCompletionHandler(Class<T> clazz, String name) {
+            this.clazz = clazz;
+            this.name = name;
+        }
+        
+        @Override
+        public List<T> onCompleted(Response response) throws Exception {
+            logResponse(response);
+            if (isStatus2xx(response)) {
+                JsonNode responseNode = mapper.readTree(response.getResponseBodyAsBytes());
+                setPagedProperties(responseNode, clazz);
+                List<T> values = new ArrayList<T>();
+                for (JsonNode node : responseNode.get(name)) {
+                    values.add(mapper.convertValue(node, clazz));
+                }
+                return values;
+            }
+            throw new ZendeskResponseException(response);
+        }
+    }
 
     protected <T> PagedAsyncCompletionHandler<List<T>> handleList(final Class<T> clazz, final String name) {
-        return new PagedAsyncCompletionHandler<List<T>>() {
-
-            @Override
-            public List<T> onCompleted(Response response) throws Exception {
-                logResponse(response);
-                if (isStatus2xx(response)) {
-                    JsonNode responseNode = mapper.readTree(response.getResponseBodyAsBytes());
-                    setPagedProperties(responseNode, clazz);
-                    List<T> values = new ArrayList<T>();
-                    for (JsonNode node : responseNode.get(name)) {
-                        values.add(mapper.convertValue(node, clazz));
-                    }
-                    return values;
+        return new PagedAsyncListCompletionHandler<T>(clazz, name);
+    }
+    
+    protected <T> PagedAsyncCompletionHandler<List<T>> handleIncrementalList(final Class<T> clazz, final String name) {
+        return new PagedAsyncListCompletionHandler<T>(clazz, name) {
+            public void setPagedProperties(JsonNode responseNode, Class<?> clazz) {
+                JsonNode node = responseNode.get(NEXT_PAGE);
+                if (node == null) {
+                    throw new NullPointerException(NEXT_PAGE + " property not found, pagination not supported" +
+                            (clazz != null ? " for " + clazz.getName() : ""));
+                }
+                JsonNode endTimeNode = responseNode.get(END_TIME);
+                if (endTimeNode == null || endTimeNode.asLong() == 0) {
+                    throw new NullPointerException(END_TIME + " property not found, incremental export pagination not supported" +
+                            (clazz != null ? " for " + clazz.getName() : ""));
+                }
+                if (endTimeNode.asLong() * 1000 > System.currentTimeMillis() - FIVE_MINUTES_IN_MS) {
```

Comment:
You could use `TimeUnit.SECONDS.toMillis(endTimeNode.asLong())` rather than the less expressive `* 1000`